packages/web3-core/types/index.d.ts

```diff
@@ -411,9 +410,16 @@ export interface LogsOptions {
 
 export type BlockNumber = string | number | BN | BigNumber | 'latest' | 'pending' | 'earliest' | 'genesis';
 
+export interface RequestArguments {
+    method: string;
+    params?: unknown;
+    [key: string]: unknown;
+}
+
 export interface AbstractProvider {
-    send(payload: JsonRpcPayload, callback: (error: Error | null, result?: JsonRpcResponse) => void): void;
     sendAsync(payload: JsonRpcPayload, callback: (error: Error | null, result?: JsonRpcResponse) => void): void;
+    send?(payload: JsonRpcPayload, callback: (error: Error | null, result?: JsonRpcResponse) => void): void;
+    request?(args: RequestArguments): Promise<unknown>;
```

Comment:
Purely out of curiosity, what's the behavior of `unknown` here when assigning the result? Like, do you have to cast it or it "just works".

Reply:
Ah yes thanks I'll change to `any` since `unknown` will force the user to type cast the result and I don't think we need to enforce that.